### PR TITLE
feat: config editor

### PR DESCRIPTION
**Pull request by MasterKenth**
Added ability to edit (some) settings in frontend; very cool 😎.
image

Not all settings are migrated from `.env` to db yet - only `FIXED_PIN`, `OPENAI_API_KEY`, `BREVO_API_URL` and `BREVO_API_KEY` right now. Migration requires changing a lot of code since settings fetching is now async/db-dependent.

# BREAKING CHANGES
* `FIXED_PIN` can no longer be `None` - instead a blank value `''` means "disabled".
* `MAIL_CLIENT` no longer does anything - instead Brevo is always used as long as `BREVO_API_KEY` is not blank (`''`).

### CODE REVIEW

File: fai-rag-app/fai-backend/fai_backend/dependencies.py [hunk 1]
```diff
@@ -71,7 +72,9 @@ def create_menus() -> list:
                 *qa_menu(user_permissions=permissions),
                 *document_menu_items(user_permissions=permissions),
                 *assistant_menu(user_permissions=permissions),
-                *feedback_menu_items()]
+                *feedback_menu_items(),
+                *mock_menu(user_permissions=permissions),
```
Comment (dannil76) [inline fai-rag-app/fai-backend/fai_backend/dependencies.py:76]: behöver vi mock_menu? Togs bort för att den inte gav något vettigt i dagsläget. :)

File: fai-rag-app/fai-backend/fai_backend/settings/models.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from typing import Dict
+
+from pydantic import BaseModel
+
+SettingsDict = Dict[str, bool | float | int | str]  # Note: order of types is important!
```
Comment (dannil76) [inline fai-rag-app/fai-backend/fai_backend/settings/models.py:5]: Vore fint att få veta varför? 👯‍♂️ :)

**Review by dannil76 — APPROVED** [review]

File: fai-rag-app/fai-backend/fai_backend/mail/client.py [hunk 3]
```diff
@@ -7,48 +7,55 @@
...
 
-    return {
-        'console': ConsoleMailClient,
-        'brevo': BrevoMailClient,
-    }[settings.MAIL_CLIENT]()
+    async def send_mail(self, payload: EmailPayload) -> bool:
+        payload = BrevoPayload.model_validate(payload.model_dump())
+        settings_service = SettingsServiceFactory().get_service()
+        url = await settings_service.get_value(SettingKey.BREVO_API_URL)
+        key = await settings_service.get_value(SettingKey.BREVO_API_KEY)
```
Comment (zilaei) [inline fai-rag-app/fai-backend/fai_backend/mail/client.py:38]: `send_mail` raises `requests.exceptions.InvalidHeader: Header part (SecretStr('**********')) from ('api-key', SecretStr('**********')) must be of type str or bytes, not <class 'pydantic.types.SecretStr'>`

Might be cause by my test environment. There is a value set for `FIXED_PIN` in my `.env`.

Comment (MasterKenth) [reply fai-rag-app/fai-backend/fai_backend/mail/client.py:38]: This probably also applies to `OPENAI_API_KEY` and any other `SecretStr` env var.

**Pull request merged**